dGame/dComponents/InventoryComponent.cpp

```diff
@@ -989,6 +898,82 @@ void InventoryComponent::UnEquipItem(Item* item) {
 	}
 }
 
+void InventoryComponent::HandlePossession(Item* item) {
+	auto* characterComponent = m_Parent->GetComponent<CharacterComponent>();
+	if (!characterComponent) return;
+
+	auto* possessorComponent = m_Parent->GetComponent<PossessorComponent>();
+	if (!possessorComponent) return;
+
+	// don't do anything if we are busy dismounting
+	if (possessorComponent->GetIsDismounting()) return;
+
+	// check to see if we are already mounting something
+	auto* currentlyPossessedEntity = EntityManager::Instance()->GetEntity(possessorComponent->GetPossessable());
+	auto currentlyPossessedItem = possessorComponent->GetMountItemID();
+
+	if (currentlyPossessedItem) {
+		if (currentlyPossessedEntity) possessorComponent->Dismount(currentlyPossessedEntity);
+		return;
+	}
+	// check again cause sometimes we need to
+	if (possessorComponent->GetIsDismounting()) return;
+	GameMessages::SendSetStunned(m_Parent->GetObjectID(), eStunState::PUSH, m_Parent->GetSystemAddress(), LWOOBJID_EMPTY, true, false, true, false, false, false, false, true, true, true, true, true, true, true, true, true);
+
+	// set the mount Item ID so that we know what were handling
+	possessorComponent->SetMountItemID(item->GetId());
+	GameMessages::SendSetMountInventoryID(m_Parent, item->GetId(), UNASSIGNED_SYSTEM_ADDRESS);
+
+	// create entity to mount
+	auto startRotation = m_Parent->GetRotation();
+
+	EntityInfo info{};
+	info.lot = item->GetLot();
+	info.pos = m_Parent->GetPosition();
+	info.rot = startRotation;
+	info.spawnerID = m_Parent->GetObjectID();
+
+	auto* mount = EntityManager::Instance()->CreateEntity(info, nullptr, m_Parent);
+
+	// check to see if the mount is a vehicle, if so, flip it
+	auto* vehicleComponent = mount->GetComponent<VehiclePhysicsComponent>();
+	if (vehicleComponent) {
+		auto angles = startRotation.GetEulerAngles();
+		angles.x -= PI;
+		startRotation = NiQuaternion::FromEulerAngles(angles);
+		mount->SetRotation(startRotation);
+		// we're pod racing now
+		characterComponent->SetIsRacing(true);
+	}
+
+	// Setup the destroyable stats
+	auto* destroyableComponent = mount->GetComponent<DestroyableComponent>();
+	if (destroyableComponent) {
+		destroyableComponent->SetIsSmashable(false);
+		destroyableComponent->SetIsImmune(true);
+	}
+
+	// mount it
+	auto* possessableComponent = mount->GetComponent<PossessableComponent>();
+	if (possessableComponent) {
+		possessableComponent->SetIsItemSpawned(true);
+		possessableComponent->SetPossessor(m_Parent->GetObjectID());
+		// possessor stuff
+		possessorComponent->SetPossessable(mount->GetObjectID());
+		possessorComponent->SetPossessableType(possessableComponent->GetPossessionType());
+	}
+
+	GameMessages::SendSetJetPackMode(m_Parent, false);
+
+	EntityManager::Instance()->ConstructEntity(mount);
+	EntityManager::Instance()->SerializeEntity(mount);
```

Comment:
This should not be needed?

Reply:
i'll check again, but the mount was not controllable without that serilization

Reply:
you are correct, will remove